### ISSUE TITLE
No Valid Logs Found

### ISSUE THREAD

**Issue by bryan1anderson** [RobertGummesson/BuildTimeAnalyzer-for-Xcode#30]
![screen shot 2016-05-23 at 11 40 33 am](https://cloud.githubusercontent.com/assets/7976521/15479604/2c7c5110-20de-11e6-9301-7005f8949b2e.png)

Everything seems to be working. I hit build, Build Time Analyzer updates and shows that a build is in progress, but it ends with Status: "No valid logs found"

I have verified that the two Swift flags have been added to the correct target, but still nada

![screen shot 2016-05-23 at 12 05 13 pm](https://cloud.githubusercontent.com/assets/7976521/15479700/a717cefe-20de-11e6-9020-2b636f040b6c.png)

**Comment by RobertGummesson** [comment]
Do you have more than one target in your project? If so, was it added to the same target as you are building?

**Comment by bryan1anderson** [comment]
@RobertGummesson no, no other targets. Single target app and they were for sure added to the build settings of the correct target

**Comment by RobertGummesson** [comment]
Just to rule out some possible scenarios, could you try this on another project and see if you have the same issue there?

**Comment by RobertGummesson** [comment]
@bryan1anderson I just managed to experience a similar issue. In my case, I have been playing around with timezones and Xcode got confused when I had log files made in the future and refused to output the build time bodies.

I deleted all folders in the derivedData folder. After that it worked again.

**Comment by gprasant** [comment]
@RobertGummesson I have the same problem. 
When Im building, I see this 

![building](https://cloud.githubusercontent.com/assets/212911/15683974/dcb5cdb8-2719-11e6-9b94-5dfaabfc3314.png)

After the build completes, I see this 

![no logs](https://cloud.githubusercontent.com/assets/212911/15683967/d50047a6-2719-11e6-82a5-20eb480645ef.png)

I followed your advice of deleting derived data. I tried running it on another simpler "New Project" and it worked. But it doesn't work on a larger project that I spend most of my time on. Do you have a troubleshooting list ? 

Thanks for this plugin, It will be hugely useful to reduce my build times.

**Comment by gprasant** [comment]
In my case, there is a XCode 'project' and there are multiple XCode 'targets' under that project. I have the settings on the 'target' for my app. But it had the above behavior.

**Comment by bryan1anderson** [comment]
@gprasant I'm afraid deleting derived data hasn't worked for me either

**Comment by gprasant** [comment]
🏆 . I went through my build settings more thoroughly and changed things and got the profiling to work.

**Comment by RobertGummesson** [comment]
@gprasant - Great. What was missing?

@bryan1anderson 

It would be useful to narrow down the scope of possible causes. Since I can't reproduce this myself, I will unfortunately need to rely on someone else to give me a hand debugging this.

If I had to guess, the problem is that the productName is not matching workspace name (line 46, CMXcodeWorkspace.swift). If not there, it would be interesting to know where it goes wrong.

To debug this, first build the project you have an issue with to ensure that the build log is created. Then do the following steps:
- Download the plug-in source code
- Configure the plug-in to run with Xcode. Edit Scheme. Then select the Run tab. Then under the Info tab, select Xcode as the Debug Executable.
- Run the project. (This launches a second instance of Xcode)
- Open up the project you have an issue with.
- Do Ctrl-Shift B top open up the Build Time Analyzer.

Something is most likely going wrong in the function _logTextForProduct()_ in CMLogProcessor.swift.

**Comment by bryan1anderson** [comment]
@Gprasant any details that could be helpful to me?

**Comment by bryan1anderson** [comment]
I'll get on that as soon as I can today and let you know what I find

**Comment by RobertGummesson** [comment]
Actually, before debugging the code, it may be worth just checking that the build times are being created properly by Xcode. The steps for manually looking at these are detailed in [this post](https://medium.com/swift-programming/speeding-up-slow-swift-build-times-922feeba5780#.5j95gja8y).

**Comment by gprasant** [comment]
my theory is that the compiler flags were not applied correctly at the first try. My build settings look like this : 

![screen shot 2016-05-31 at 11 43 12 am](https://cloud.githubusercontent.com/assets/212911/15686465/4e1e445c-2725-11e6-9ec3-489d10a3f5ae.png)


When I added the compiler flags ( -Xfrontend -debug-time-function-bodies ), it did not get added to the leaf nodes in that tree (The ones that say 'Any Architecture | Any SDK' ). 
In my second try, I had to make sure the flags were set on the leaf nodes as well. It worked after that. 
@RobertGummesson is there a chatroom for this repo ?

**Comment by RobertGummesson** [comment]
Thanks for sharing. ...and no, there is no specific chatroom set up for this repo.

**Comment by Lascorbe** [comment]
Having the same issue on a big project here, it: 
- It's a mix of swift and objc. 
- Uses CocoaPods. 
- Workspace and Target are the same name.

Any clue about it? Or do you know how can I debug it?

**Comment by RobertGummesson** [comment]
Hi @Lascorbe 

I've used it on projects using CocoaPods and a mix of Swift and Obj-C so at least that can be ruled out as a possible cause. To summarize some of the points you can try:

- Ensure that the compiler flags are added to the appropriate nodes (as per @gprasant's post).
- Ensure that what you are building matches the target you added the flags to.
- Check if the build times are being created properly by Xcode (see my last comment from May 31)
- Clean the project and then rebuild while the window is open.

**Comment by Lascorbe** [comment]
Hey @RobertGummesson!

- [ ]  Ensure that the compiler flags are added to the appropriate nodes (as per @gprasant's post).
- [ ]  Ensure that what you are building matches the target you added the flags to.
- [ ]  Check if the build times are being created properly by Xcode (see my last comment from May 31)
- [ ] Clean the project and then rebuild while the window is open.

Looks like everything is fine but... **I'm seeing the same as gprasant [here](https://github.com/RobertGummesson/BuildTimeAnalyzer-for-Xcode/issues/30#issuecomment-222759097)**

I'm gonna try different flags configurations and see if I can get it working.

Thank you for your help :)

**Comment by preraksola** [comment]
I have the same behaviour. I tried deleting the DerivedData folder contents but of no use.
I have multiple targets in my application, and I added the flags in Build Settings of my project (not targets). In targets I have `($inherited)` so they all got the flags.

**Comment by evercry** [comment]
I have the same problem, in my case Derived data option is Relative so that BTA is not working and then I've changed this option to default that BTA is working, but I need to set the derived data option is Relative. Any idea?

**Comment by d1248966459** [comment]
i have the same problem

**Comment by klausbussekw** [comment]
Actually I ran into this because I had globally set the derived data folder to a custom location… so check your derived data path in your XCode > Preferences > Locations

**Comment by kuriishu27** [comment]
In my case I had my Unit and UI Tests in the App's Target. 
By setting only the App as a target, it should work.

![build-time](https://user-images.githubusercontent.com/18484997/29516149-2a8bbd7e-8667-11e7-9e10-d8b45c57ece0.jpg)

**Comment by threat70** [comment]
this issue backed again in 1.0.7 version. in 1.0.6 all ok. Tested with Xcode 9 GM

**Comment by RobertGummesson** [comment]
@threat70 - Thanks for reporting. I just fixed it in v1.0.8.

**Comment by threat70** [comment]
@RobertGummesson wow, so fast. Thank you :)

**Comment by davidbjames** [comment]
Late to this party. For me, the solution was to restart BuildTimeAnalyzer and the logs show up.

**Comment by kmcginnes** [comment]
I was having these issues and decided to start debugging to see if I could figure out what the issue is. I haven't pin pointed the exact cause, but it does seem to be a timing/threading issue.

I put breakpoints in a few places and as I was stepping through the log processing bit I noticed it was working where it didn't work before. If I turn off the breakpoints it doesn't work again.

Here are the breakpoints that I used:

<img width="1792" alt="Screen Shot 2019-11-25 at 8 59 56 AM" src="https://user-images.githubusercontent.com/212862/69553886-87581900-0f66-11ea-9348-08b8c5532c21.png">

If I figure out which of the background threads are stomping on which, I'll report back.
